src/sentry/snuba/discover.py

```diff
@@ -894,24 +900,36 @@ def histogram_query(
     )
 
     key_column = None
+    array_column = None
+    histogram_function = None
     conditions = []
     if len(fields) > 1:
-        key_column = "array_join(measurements_key)"
+        histogram_type = check_histogram_fields(fields)
+        if histogram_type == "measurements":
+            key_column = "array_join(measurements_key)"
+            array_column = "measurements"
+            histogram_function = get_measurement_name
+        elif histogram_type == "op_breakdown":
+            key_column = "array_join(span_op_breakdowns_key)"
+            array_column = "span_op_breakdowns"
+            histogram_function = get_span_op_breakdown_name
```

Comment:
Once you make `check_histogram_fields` return the full name, `array_column` becomes unnecessary since you can just use `histogram_type` in its place.

Reply:
Ok will do